libs/auth/src/inserted_smart_card_auth_with_memory_card.ts

```diff
@@ -49,70 +45,56 @@ export class InsertedSmartCardAuthWithMemoryCard
 {
   private authStatus: InsertedSmartCardAuth.AuthStatus;
   private readonly card: Card;
-  private config: InsertedSmartCardAuthConfig;
+  private readonly config: InsertedSmartCardAuthConfig;
 
   constructor({
     card,
     config,
   }: {
     card: Card;
-    config: Omit<
-      InsertedSmartCardAuthConfig,
-      'electionDefinition' | 'precinctSelection'
-    >;
+    config: InsertedSmartCardAuthConfig;
   }) {
     this.authStatus = InsertedSmartCardAuth.DEFAULT_AUTH_STATUS;
     this.card = card;
     this.config = config;
-
-    setInterval(
-      async () => {
-        try {
-          const newCardSummary = await this.card.readSummary();
-          this.updateAuthStatus({
-            type: 'read_card',
-            cardSummary: newCardSummary,
-          });
-        } catch {
-          // Swallow errors so that they don't crash the auth instance and containing backend
-        }
-      },
-      CARD_POLLING_INTERVAL_MS,
-      true
-    );
   }
 
-  getAuthStatus(): InsertedSmartCardAuth.AuthStatus {
+  async getAuthStatus(
+    machineState: InsertedSmartCardAuthMachineState
+  ): Promise<InsertedSmartCardAuth.AuthStatus> {
+    await this.checkCardReaderAndUpdateAuthStatus(machineState);
     return this.authStatus;
   }
 
-  checkPin({ pin }: { pin: string }): void {
-    this.updateAuthStatus({ type: 'check_pin', pin });
+  async checkPin(
+    machineState: InsertedSmartCardAuthMachineState,
+    input: { pin: string }
+  ): Promise<void> {
+    await this.checkCardReaderAndUpdateAuthStatus(machineState);
+    this.updateAuthStatus(machineState, { type: 'check_pin', pin: input.pin });
   }
 
-  readCardData<T>({
-    schema,
-  }: {
-    schema: z.ZodSchema<T>;
-  }): Promise<Result<Optional<T>, SyntaxError | z.ZodError | Error>> {
-    return this.card.readLongObject(schema);
+  readCardData<T>(
+    machineState: InsertedSmartCardAuthMachineState,
+    input: { schema: z.ZodSchema<T> }
+  ): Promise<Result<Optional<T>, SyntaxError | z.ZodError | Error>> {
```

Comment:
Though I'm not doing this yet since this implementation will be short lived, I could call `this.checkCardReaderAndUpdateAuthStatus(machineState)` and then verify that the user is logged in as a sys admin and that there's a card in the card reader